sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamJoinRel.java

```diff
@@ -198,6 +195,9 @@ private boolean isSideInputJoin() {
       if ((leftRows.isBounded() == PCollection.IsBounded.BOUNDED
               && rightRows.isBounded() == PCollection.IsBounded.BOUNDED)
           || (leftRows.isBounded() == UNBOUNDED && rightRows.isBounded() == UNBOUNDED)) {
+        verifySupportedTrigger(leftRows);
```

Comment:
@akedin I think you added this - can you double check my thinking that it is safe to move into here?

Reply:
I think so, this is the CoGBK path where we care about the windowing+triggering. The other path is side-input where it should work anyhow, I think